Jetcaster/wear/src/main/java/com/example/jetcaster/ui/episode/EpisodeScreen.kt

```diff
@@ -0,0 +1,282 @@
+import com.google.android.horologist.compose.material.AlertDialog
+import com.google.android.horologist.compose.material.Button
+import com.google.android.horologist.compose.material.ListHeaderDefaults
+import com.google.android.horologist.compose.material.ResponsiveListHeader
+import com.google.android.horologist.media.ui.screens.entity.EntityScreen
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+
+@Composable fun EpisodeScreen(
+    onPlayButtonClick: () -> Unit,
+    onDismiss: () -> Unit,
+    modifier: Modifier = Modifier,
+    episodeViewModel: EpisodeViewModel = hiltViewModel()
+) {
+    val uiState by episodeViewModel.uiState.collectAsStateWithLifecycle()
+
+    EpisodeScreen(
+        uiState = uiState,
+        onPlayButtonClick = onPlayButtonClick,
+        modifier = modifier,
+        onPlayEpisode = episodeViewModel::onPlayEpisode,
+        onAddToQueue = episodeViewModel::addToQueue,
+        onDismiss = onDismiss,
+    )
+}
+
+@Composable
+fun EpisodeScreen(
+    uiState: EpisodeScreenState,
+    onPlayButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    onPlayEpisode: (PlayerEpisode) -> Unit,
+    onAddToQueue: (PlayerEpisode) -> Unit,
+    onDismiss: () -> Unit
+) {
+    val columnState = rememberResponsiveColumnState(
+        contentPadding = padding(
+            first = ScalingLazyColumnDefaults.ItemType.Text,
+            last = ScalingLazyColumnDefaults.ItemType.Chip
+        )
+    )
+    ScreenScaffold(
+        scrollState = columnState,
+        modifier = modifier
+    ) {
+        when (uiState) {
+            is EpisodeScreenState.Loaded -> {
+                val author = uiState.episode.episode.author
+                val duration = uiState.episode.episode.duration
+                val published = uiState.episode.episode.published
+                val summary = uiState.episode.episode.summary
+                val title = uiState.episode.episode.title
+
+                EntityScreen(
+                    columnState = columnState,
+                    headerContent = {
+                        ResponsiveListHeader(
+                            contentPadding = ListHeaderDefaults.firstItemPadding()
+                        ) {
+                            Text(
+                                text = title,
+                                maxLines = 1,
+                                overflow = TextOverflow.Ellipsis
+                            )
+                        }
+                    },
+                    buttonsContent = {
+                        LoadedButtonsContent(
+                            episode = uiState.episode,
+                            onPlayButtonClick = onPlayButtonClick,
+                            onPlayEpisode = onPlayEpisode,
+                            onAddToQueue = onAddToQueue
+                        )
+                    },
+                    content = {
+                        if (!author.isNullOrEmpty()) {
```

Comment:
Consider defining the content in a separate function. It's ok as-s but it would improve readability.

Reply:
thanks, done